recipes/Myst/myst_prepare.py

```diff
@@ -0,0 +1,560 @@
+
+    # Skip files longer than 30s
+    if duration > 30.0:
+        return None
+
+    return MystRow(
+        snt_id=snt_id,
+        spk_id=spk_id,
+        duration=duration,
+        file_path=audio_file,
+        words=wrds,
+    )
+
+
+class _ASRBackend:
+    """Thin wrapper around Faster-Whisper or OpenAI Whisper.
+
+    Selects the available backend at construction time and exposes
+    a single `transcribe(path)` method used for WER filtering.
+    """
+    def __init__(self, model_name: str = "small", device: Optional[str] = None):
+        if _HAS_FASTER:
+            self.kind = "faster"
+            self.model = WhisperModel(model_name, device=device or "auto", compute_type="float32")
+        elif _HAS_OPENAI:
+            self.kind = "openai"
+            self.model = whisper.load_model(model_name, device=device or None)  # type: ignore
+        else:
+            raise RuntimeError("No ASR backend. Install faster-whisper or openai-whisper.")
+
+    def transcribe(self, path: str) -> str:
+        """Transcribe an audio file to text.
+
+        Arguments
+        ---------
+        path : str
+            Path to the audio file.
+
+        Returns
+        -------
+        str
+            Normalized ASR hypothesis.
+        """
+        if self.kind == "faster":
+            segs, _ = self.model.transcribe(path, task="transcribe", language="en")
+            hyp = " ".join(s.text for s in segs).strip()
+        else:
+            out = self.model.transcribe(path, task="transcribe", language="en")
+            hyp = out.get("text", "").strip()
+        return _normalize_text(hyp)
+
+
+def _apply_wer_filter(rows: List[MystRow], threshold: float, asr_model: str, device: Optional[str]) -> List[MystRow]:
+    """Filter rows using zero-shot ASR and WER.
+
+    Runs an ASR model (whisper) on each utterance and keeps only those whose
+    word error rate (WER) is below or equal to the given threshold. This is done
+    to remove transcription errors present in the dataset.
+
+    Arguments
+    ---------
+    rows : list of MystRow
+        Input rows to filter.
+    threshold : float
+        Maximum allowed WER (e.g., 0.3 for 30%).
+    asr_model : str
+        Whisper model size/name to use for ASR.
+    device : str, optional
+        Device for ASR inference (e.g., 'cpu', 'cuda').
+
+    Returns
+    -------
+    list of MystRow
+        The subset of rows that pass the WER filter.
+    """
+    if threshold is None:
+        return rows
+    if not _HAS_JIWER:
+        raise RuntimeError("jiwer is required for WER filtering. pip install jiwer")
+    asr = _ASRBackend(asr_model, device=device)
+    kept: List[MystRow] = []
+    for r in tqdm(rows, desc="WER filtering"):
+        hyp = asr.transcribe(r.file_path)
+        ref = _normalize_text(r.words)
+        hyp = _normalize_text(hyp)
+        score = wer(ref, hyp)
+        if score <= threshold:
+            kept.append(r)
+    return kept
```

Comment:
I think I would prefer doing this in two files. One file does the basic data prep, and then the other file does the WER scoring and filtering. This way we keep things simple.